tls/s2n_ktls.c

```diff
@@ -90,6 +90,16 @@ static S2N_RESULT s2n_ktls_validate(struct s2n_connection *conn, s2n_ktls_mode k
     bool may_renegotiate = may_receive_hello_request && config->renegotiate_request_cb;
     RESULT_ENSURE(!may_renegotiate, S2N_ERR_KTLS_RENEG);
 
+    /* Prevent kTLS from being enabled on connections that might be serialized.
+     *
+     * When kTLS is enabled the socket takes over tracking sequence numbers. This means
+     * s2n-tls doesn't know the sequence numbers, a requirement for serializing
+     * without errors. This might change in the future, but for now, we don't allow
+     * kTLS if the serialization version is set.
```

Comment:
I'd maybe phrase this a little differently to be clear it's a choice to simplify implementation rather than a hard blocker:
```suggestion
     * When kTLS is enabled the socket takes over tracking sequence numbers.
     * Because serialization requires the current sequence number, we would need to
     * call getsockopt to retrieve the current sequence numbers. That would complicate
     * the serialization implementation. For now, do not support kTLS with serialization.
```